### DISCUSSION TITLE
Unexpected Publisher Confirmation Behavior

### DISCUSSION THREAD

**Discussion by IlyaGav** [rabbitmq/rabbitmq-dotnet-client#1089] [General]
I have two queues: `queue1` and `queue2`.  `queue2` has two arguments: `x-max-length: 10` and `x-overflow: reject-publish`.
I want to move messages from `queue1` to `queue2` and it is important for me that no messages get lost.  
Here is my code

```
_model = _connection.CreateModel();

var consumer = new EventingBasicConsumer(_model);
consumer.Received += TransferTargets;

_model.ConfirmSelect();
_model.BasicAcks += OnAcknowledged;
_model.BasicNacks += OnNotAcknowledged;

_model.BasicConsume("queue1", false, consumer);
```  
```
private void TransferTargets(object sender, BasicDeliverEventArgs e)
{
    var body = e.Body.ToArray();
    var props = _model.CreateBasicProperties();
    _model.BasicPublish("queue2", "", props, body);
}
```
```
private void OnAcknowledged(object model, BasicAckEventArgs args)
{
    _model.BasicAck(args.DeliveryTag, false);
}
```
```
private void OnNotAcknowledged(object model, BasicNackEventArgs args)
{
    _model.BasicReject(args.DeliveryTag, true);
}
```
The `OnAcknowledged` method is called when a message has been delivered to `queue2`.
The `OnNotAcknowledged` method is called when a message has not been delivered to` queue2`.

Thus, if at first there were `100` messages in` queue1` and `0` in `queue2`, then after starting something like this happens: `10` messages are added to `queue2` (since there is such a limitation on it), and `queue1` can be `95`, that is, the `OnAcknowledged` method was called in total `5 times`, although all `10` messages were successfully delivered.

Is this intended behavior or am I doing something wrong?

Environment:  
.Net Core version: 3.1   
RabbitMQ Server version: 3.8.11  
RabbitMQ.Client version: 6.2.1  

Sorry, I don't know English well, so I use a translator

**Comment by lukebakken** [comment]
More than likely, the `multiple` flag was set on some of the confirms. If so, this means that multiple messages were confirmed at the same time.
